Core/State.cs

```diff
@@ -1,73 +1,30 @@
-﻿using Core.TimeSeries;
-using System;
+﻿using System;
 using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
+using Newtonsoft.Json;
 
 namespace Core
 {
-	public static class State
+	public class State
 	{
-		private static readonly Dictionary<string, Func<Transaction, bool>> categoryFilters = new Dictionary<string, Func<Transaction, bool>>();
+		public static State Instance { get; set; } = new State(new List<Category>(), new SortedSet<Transaction>());
 
-		public static SortedSet<Transaction> Transactions { get; } = new SortedSet<Transaction>();
+		public SortedSet<Transaction> Transactions { get; }
 
-		public static HashSet<string> Categories => new HashSet<string>(categoryFilters.Keys);
+		public List<Category> Categories { get; }
+		
+		[JsonIgnore]
+		public Dictionary<string, Func<Transaction, bool>> CategoryFilters { get; }
 
-		public static event Action OnStateUpdated = () => { };
+		// todo: move to IReadOnlySet in .NET 5
```

Comment:
Codacy found an issue: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=5889469)